service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/ControlledAzureDatabaseResource.java

```diff
@@ -0,0 +1,317 @@
+            getWorkspaceId(),
+            flightBeanBag.getResourceDao()),
+        cloudRetry);
+    flight.addStep(
+        new CreateAzureDatabaseStep(
+            flightBeanBag.getAzureConfig(),
+            flightBeanBag.getCrlService(),
+            this,
+            flightBeanBag.getLandingZoneApiDispatch(),
+            flightBeanBag.getSamService(),
+            flightBeanBag.getWorkspaceService(),
+            getWorkspaceId(),
+            new KubernetesClientProviderImpl(),
+            flightBeanBag.getResourceDao()),
+        cloudRetry);
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public void addDeleteSteps(DeleteControlledResourcesFlight flight, FlightBeanBag flightBeanBag) {
+    flight.addStep(
+        new DeleteAzureDatabaseStep(
+            flightBeanBag.getAzureConfig(),
+            flightBeanBag.getCrlService(),
+            this,
+            flightBeanBag.getLandingZoneApiDispatch(),
+            flightBeanBag.getSamService(),
+            flightBeanBag.getWorkspaceService(),
+            getWorkspaceId()),
+        RetryRules.cloud());
+  }
+
+  // Azure resources currently do not implement updating.
+  @Override
+  public void addUpdateSteps(UpdateResourceFlight flight, FlightBeanBag flightBeanBag) {}
+
+  public ApiAzureDatabaseResource toApiResource() {
+    return new ApiAzureDatabaseResource()
+        .metadata(super.toApiMetadata())
+        .attributes(toApiAttributes());
+  }
+
+  private ApiAzureDatabaseAttributes toApiAttributes() {
+    return new ApiAzureDatabaseAttributes().databaseName(getDatabaseName());
+  }
+
+  @Override
+  public String attributesToJson() {
+    return DbSerDes.toJson(
+        new ControlledAzureDatabaseAttributes(
+            getDatabaseName(), getDatabaseOwner(), k8sNamespace, getRegion()));
+  }
+
+  @Override
+  public ApiResourceAttributesUnion toApiAttributesUnion() {
+    ApiResourceAttributesUnion union = new ApiResourceAttributesUnion();
+    union.azureDatabase(toApiAttributes());
+    return union;
+  }
+
+  @Override
+  public void validate() {
+    super.validate();
+    if (getResourceType() != WsmResourceType.CONTROLLED_AZURE_DATABASE
+        || getResourceFamily() != WsmResourceFamily.AZURE_DATABASE
+        || getStewardshipType() != StewardshipType.CONTROLLED) {
+      throw new InconsistentFieldsException("Expected controlled AZURE_DATABASE");
+    }
+    if (getDatabaseName() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required databaseName field for ControlledAzureDatabase.");
+    }
+    if (getDatabaseOwner() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required databaseOwner field for ControlledAzureDatabase.");
+    }
+    if (getRegion() == null) {
+      throw new MissingRequiredFieldException(
+          "Missing required region field for ControlledAzureDatabase.");
+    }
+    AzureResourceValidationUtils.validateAzureDatabaseName(getDatabaseName());
```

Comment:
Seems like k8snamespace should also be required here.

Reply:
we are not creating the namespace here, I think its existence or not is validation enough